### PR TITLE
Add deleter function to chartfilter

### PR DESCRIPTION
**Pull request by schaumb**

### CODE REVIEW

**Comment by simzer** [comment]
pls run: `project/js$ npm run prettier`

**Comment by schaumb** [comment]
After a `npm install`. Thanks!

**Comment by simzer** [comment]
The purpose of this modification is to eliminate the reinterpret cast?

**Comment by schaumb** [comment]
Yes. And remove `removeJsFunction` from interfacejs

File: src/apps/weblib/js-api/data.js [hunk 1]
```diff
@@ -261,10 +261,20 @@ export default class Data {
...
+      let callbackPtrs = [this.chart.module.addFunction(callback, "ii")];
+      let deleter = (ptr) => {
+        if (ptr !== callbackPtrs[0])
+          console.warn("Wrong pointer passed to destructor");
+        this.chart.module.removeFunction(callbackPtrs[0]);
+        this.chart.module.removeFunction(callbackPtrs[1]);
+      };
+      callbackPtrs.push(this.chart.module.addFunction(deleter, "vi"));
+      this.chart._call(this.chart.module._chart_setFilter)(
+        callbackPtrs[0],
```
Comment (simzer) [inline src/apps/weblib/js-api/data.js:273]: can be called with `(...callbackPtrs)`

**Review by simzer — DISMISSED** [review]

**Review by simzer — APPROVED** [review]

**Pull request merged**